checkstyle-checks/src/main/resources/sonatype/checkstyle-configuration.xml

```diff
@@ -0,0 +1,181 @@
+<!--
+
+    Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
+    Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
+    "Sonatype" is a trademark of Sonatype, Inc.
+
+-->
+<!DOCTYPE module PUBLIC
+    "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+<module name="Checker">
+  <property name="charset" value="UTF-8"/>
+
+  <property name="severity" value="error"/>
+
+  <property name="fileExtensions" value="java"/>
+  <module name="FileTabCharacter">
+    <property name="eachLine" value="true"/>
+  </module>
+
+  <module name="NewlineAtEndOfFile"/>
+
+  <module name="SuppressWarningsFilter" />
+
+  <module name="TreeWalker">
+    <module name="com.sonatype.checks.ClassStructureEmptyLineCheck" />
+    <module name="SuppressWarningsHolder" />
+    <module name="UnnecessaryParentheses"/>
+    <module name="EmptyLineSeparator">
+      <property name="allowMultipleEmptyLines" value="false" />
+      <property name="allowMultipleEmptyLinesInsideClassMembers" value="false" />
+      <property name="tokens" value="IMPORT, CLASS_DEF, INTERFACE_DEF, ENUM_DEF, STATIC_INIT, INSTANCE_INIT, METHOD_DEF, CTOR_DEF, VARIABLE_DEF"/>
+    </module>
+    <module name="OuterTypeFilename"/>
+    <module name="LineLength">
+      <property name="max" value="120"/>
+      <!-- Allow longer lines in comments. -->
+      <property name="ignorePattern" value="^ \*.*"/>
+    </module>
+    <module name="UnusedImports"/>
+    <module name="AvoidStarImport"/>
```

Comment:
Without customization of the module, how is this not causing violations for our existing code which makes use of star imports in accordance with the thresholds given by https://github.com/sonatype/codestyle#imports ?

Reply:
> Without customization of the module, how is this not causing violations for our existing code which makes use of star imports in accordance with the thresholds given by https://github.com/sonatype/codestyle#imports ?

The customization for this module don't accommodate our exceptions.    That being said, while I have not done an exhaustive search. the violations would be pretty rare.  In those cases of approved exceptions we can add the @SupressWarnings("AvoidStarImports") to the file.  Alternately we can remove this check.  Or we can write our own that is a little smarter than the default one.  Or we can just make this a warning.

Reply:
With IDEs collapsing the import section out of view, I personally don't mind if we keep the rule as long as there is a corresponding effort to change the documented codestyle accordingly. In the interest of finding consensus to move this forward, the smart play might be to defer that fight to another day and start off with a laxer checkstyle...

Reply:
> start off with a laxer checkstyle

Converted this to a warning rather than an error in 34eade2c0ee92b6495de7c9e2b06a4d231a913c5

Reply:
> Converted this to a warning rather than an error in 34eade2

Our rules say that the static imports will be converted to star when there are more than 10. The number for non-static imports is 100. Personally, I'm not a big fan of star imports and I would be all for not allowing them at all. But as long as our rules allow them, we should not configure checkstyle to flag them as errors or warnings.